packages/gafl-webapp-service/src/pages/order-complete/licence-details/__tests__/route.spec.js

```diff
@@ -26,149 +26,156 @@ jest.mock('../../../../constants.js', () => ({
-        await expect(getData(mockRequest)).rejects.toThrow(boomError)
-      })
 
-      it('if status posted flag is not set', async () => {
-        const mockRequest = createMockRequest({ status: { [COMPLETION_STATUS.agreed]: true } })
-        const boomError = Boom.forbidden('Attempt to access the licence information handler with no posted flag set')
-        await expect(getData(mockRequest)).rejects.toThrow(boomError)
-      })
+const getMockStatus = () => ({
+  [COMPLETION_STATUS.agreed]: 'agreed',
+  [COMPLETION_STATUS.posted]: 'posted',
+  [COMPLETION_STATUS.finalised]: 'finalised',
+  [CommonResults.OK]: 'ok',
+  [ShowDigitalLicencePages.YES]: 'yes'
+})
 
-      it('if status finalised flag is not set', async () => {
-        const mockRequest = createMockRequest({ status: { [COMPLETION_STATUS.agreed]: true, [COMPLETION_STATUS.posted]: true } })
-        const boomError = Boom.forbidden('Attempt to access the licence information handler with no finalised flag set')
-        await expect(getData(mockRequest)).rejects.toThrow(boomError)
-      })
-    })
+const getMockTransaction = () => ({
+  permissions: [getMockPermission()]
+})
 
-    it('calls licenceTypeDisplay with permission and i18n catalog', async () => {
-      const i18nCatalog = Symbol('mock catalog')
-      const permission = Symbol('mock permission')
+const getMockPermission = () => ({
+  licensee: {
+    firstName: 'Turanga',
+    lastName: 'Leela',
+    obfuscatedDob: 'obfuscatedDob'
+  },
+  referenceNumber: '123456789',
+  licenceType: 'trout-and-coarse',
+  numberOfRods: '2',
+  licenceLength: '8D',
+  permit: { cost: 12 }
+})
 
-      const sampleRequest = createMockRequest({ permission, i18nCatalog })
+const getMockCatalog = overrides => ({
+  age_senior_concession: Symbol('Over 65'),
+  age_junior_concession: Symbol('Junior (13 to 16)'),
 
-      await getData(sampleRequest)
+  licence_type_1d: Symbol('1 day'),
+  licence_type_8d: Symbol('8 days'),
+  licence_type_12m: Symbol('12 months'),
+  ...overrides
+})
 
-      expect(licenceTypeDisplay).toHaveBeenCalledWith(permission, i18nCatalog)
+describe('licence-length > route', () => {
+  beforeEach(() => jest.clearAllMocks())
+
+  describe('getData', () => {
+    const getMockRequest = (statusGet = () => {}, transactionGet = () => {}, catalog = getMockCatalog()) => ({
+      cache: () => ({
+        helpers: {
+          transaction: {
+            get: async () => transactionGet
+          },
+          status: {
+            get: async () => statusGet
+          }
+        }
+      }),
+      i18n: {
+        getCatalog: () => catalog
+      }
     })
 
-    it('return value of licenceTypeDisplay is used for licenceTypeStr', async () => {
-      const licenceDisplay = Symbol('licence display')
-      licenceTypeDisplay.mockReturnValueOnce(licenceDisplay)
+    const mockRequest = getMockRequest(getMockStatus(), getMockTransaction())
```

Comment:
As I've said before, don't reuse mocks. Generate a new one for every test.

Also if the default of status and transaction get functions in the mock request generator should be getMockStatus and getMockTransaction, make them that, don't give them defaults that never get used.

Please check all through your test file for doing this - I can see other places where you're defining variables used in the test outside of the test.